[
    {
        "url": "https://api.github.com/repos/psf/black/issues/1393",
        "repository_url": "https://api.github.com/repos/psf/black",
        "labels_url": "https://api.github.com/repos/psf/black/issues/1393/labels{/name}",
        "comments_url": "https://api.github.com/repos/psf/black/issues/1393/comments",
        "events_url": "https://api.github.com/repos/psf/black/issues/1393/events",
        "html_url": "https://github.com/psf/black/pull/1393",
        "id": 614805893,
        "node_id": "MDExOlB1bGxSZXF1ZXN0NDE1Mjg1NDc3",
        "number": 1393,
        "title": "Test on Python 3.9-dev",
        "user": {
            "login": "hugovk",
            "id": 1324225,
            "node_id": "MDQ6VXNlcjEzMjQyMjU=",
            "avatar_url": "https://avatars.githubusercontent.com/u/1324225?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/hugovk",
            "html_url": "https://github.com/hugovk",
            "followers_url": "https://api.github.com/users/hugovk/followers",
            "following_url": "https://api.github.com/users/hugovk/following{/other_user}",
            "gists_url": "https://api.github.com/users/hugovk/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/hugovk/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/hugovk/subscriptions",
            "organizations_url": "https://api.github.com/users/hugovk/orgs",
            "repos_url": "https://api.github.com/users/hugovk/repos",
            "events_url": "https://api.github.com/users/hugovk/events{/privacy}",
            "received_events_url": "https://api.github.com/users/hugovk/received_events",
            "type": "User",
            "site_admin": false
        },
        "labels": [],
        "state": "closed",
        "locked": false,
        "assignee": null,
        "assignees": [],
        "milestone": null,
        "comments": 3,
        "created_at": "2020-05-08T15:14:42Z",
        "updated_at": "2020-05-09T05:43:40Z",
        "closed_at": "2020-05-08T15:19:07Z",
        "author_association": "CONTRIBUTOR",
        "active_lock_reason": null,
        "draft": false,
        "pull_request": {
            "url": "https://api.github.com/repos/psf/black/pulls/1393",
            "html_url": "https://github.com/psf/black/pull/1393",
            "diff_url": "https://github.com/psf/black/pull/1393.diff",
            "patch_url": "https://github.com/psf/black/pull/1393.patch"
        },
        "body": "Python 3.9 is currently in alpha with a beta due in a couple of weeks (2020-05-18).\r\n\r\n* https://www.python.org/dev/peps/pep-0596/#schedule\r\n\r\nThis PR tests `3.9-dev` on Travis CI, currently Python 3.9.0a6+.\r\n\r\nHowever, it's failing, so as a pre-release I put it in the `allow_failures` so it won't turn the whole CI red.\r\n\r\nThe error is:\r\n```\r\n======================================================================\r\nERROR: tests.test_black (unittest.loader._FailedTest)\r\n----------------------------------------------------------------------\r\nImportError: Failed to import test module: tests.test_black\r\nTraceback (most recent call last):\r\n  File \"/opt/python/3.9-dev/lib/python3.9/unittest/loader.py\", line 436, in _find_test_path\r\n    module = self._get_module_from_name(name)\r\n  File \"/opt/python/3.9-dev/lib/python3.9/unittest/loader.py\", line 377, in _get_module_from_name\r\n    __import__(name)\r\n  File \"/home/travis/build/hugovk/black/tests/test_black.py\", line 20, in <module>\r\n    import black\r\n  File \"/home/travis/build/hugovk/black/black.py\", line 140, in <module>\r\n    TMatchResult = TResult[Index]\r\n  File \"/opt/python/3.9-dev/lib/python3.9/typing.py\", line 244, in inner\r\n    return func(*args, **kwds)\r\n  File \"/opt/python/3.9-dev/lib/python3.9/typing.py\", line 704, in __getitem__\r\n    arg = arg[subargs]\r\n  File \"/opt/python/3.9-dev/lib/python3.9/typing.py\", line 244, in inner\r\n    return func(*args, **kwds)\r\n  File \"/opt/python/3.9-dev/lib/python3.9/typing.py\", line 695, in __getitem__\r\n    _check_generic(self, params)\r\n  File \"/opt/python/3.9-dev/lib/python3.9/typing.py\", line 194, in _check_generic\r\n    raise TypeError(f\"{cls} is not a generic class\")\r\nTypeError: black.Err[black.CannotTransform] is not a generic class\r\n```\r\n\r\nhttps://travis-ci.org/github/hugovk/black/jobs/684724917",
        "performed_via_github_app": null,
        "score": 1.0
    },
    {
        "url": "https://api.github.com/repos/psf/black/issues/627",
        "repository_url": "https://api.github.com/repos/psf/black",
        "labels_url": "https://api.github.com/repos/psf/black/issues/627/labels{/name}",
        "comments_url": "https://api.github.com/repos/psf/black/issues/627/comments",
        "events_url": "https://api.github.com/repos/psf/black/issues/627/events",
        "html_url": "https://github.com/psf/black/pull/627",
        "id": 385803446,
        "node_id": "MDExOlB1bGxSZXF1ZXN0MjM0NjkwNzY3",
        "number": 627,
        "title": "Add CORS support to blackd",
        "user": {
            "login": "tuchandra",
            "id": 11623486,
            "node_id": "MDQ6VXNlcjExNjIzNDg2",
            "avatar_url": "https://avatars.githubusercontent.com/u/11623486?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/tuchandra",
            "html_url": "https://github.com/tuchandra",
            "followers_url": "https://api.github.com/users/tuchandra/followers",
            "following_url": "https://api.github.com/users/tuchandra/following{/other_user}",
            "gists_url": "https://api.github.com/users/tuchandra/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/tuchandra/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/tuchandra/subscriptions",
            "organizations_url": "https://api.github.com/users/tuchandra/orgs",
            "repos_url": "https://api.github.com/users/tuchandra/repos",
            "events_url": "https://api.github.com/users/tuchandra/events{/privacy}",
            "received_events_url": "https://api.github.com/users/tuchandra/received_events",
            "type": "User",
            "site_admin": false
        },
        "labels": [],
        "state": "closed",
        "locked": false,
        "assignee": null,
        "assignees": [],
        "milestone": null,
        "comments": 3,
        "created_at": "2018-11-29T15:56:24Z",
        "updated_at": "2018-12-10T14:59:05Z",
        "closed_at": "2018-12-10T08:15:58Z",
        "author_association": "CONTRIBUTOR",
        "active_lock_reason": null,
        "draft": false,
        "pull_request": {
            "url": "https://api.github.com/repos/psf/black/pulls/627",
            "html_url": "https://github.com/psf/black/pull/627",
            "diff_url": "https://github.com/psf/black/pull/627.diff",
            "patch_url": "https://github.com/psf/black/pull/627.patch"
        },
        "body": "See #622. This is my first time contributing to a project that uses pipenv, so please let me know if there's something I did wrong.\r\n\r\nAdditionally, I'm not sure how to add a test for this. I'd expect it to test that a request sent from a client on a different host works correctly, but I read through the docs for the aiohttp [TestClient](https://docs.aiohttp.org/en/stable/testing.html?highlight=testing#test-client) and couldn't come up with anything. Setting the host param in the constructor for TestClient `async with TestClient(TestServer(app), host=some_string)` gives a TypeError about the host parameter being invalid.",
        "performed_via_github_app": null,
        "score": 1.0
    }
]